INSTALL.md

```diff
@@ -0,0 +1,315 @@
+contents of the buffer are directly uploaded to Manta in a retryable form.
+* `manta.client_encryption` (**MANTA_CLIENT_ENCRYPTION**)
+Boolean indicating if client-side encryption is enabled.
+* `manta.encryption_key_id` (**MANTA_CLIENT_ENCRYPTION_KEY_ID**)
+Unique ID of the client-side encryption key being used. It must be in US-ASCII
+using only printable characters and with no whitespace.
+* `manta.encryption_algorithm` (**MANTA_ENCRYPTION_ALGORITHM**)
+The client-side encryption algorithm used to encrypt and decrypt data. Valid
+values are listed in [SupportedCipherDetails](java-manta-client/src/main/java/com/joyent/manta/client/crypto/SupportedCipherDetails.java#L26).
+* `manta.permit_unencrypted_downloads` (**MANTA_UNENCRYPTED_DOWNLOADS**)
+Boolean indicating that unencrypted files can be downloaded when client-side
+encryption is enabled.
+* `manta.encryption_auth_mode` (**MANTA_ENCRYPTION_AUTH_MODE**)
+[EncryptionAuthenticationMode](java-manta-client/src/main/java/com/joyent/manta/config/EncryptionAuthenticationMode.java)
+enum type indicating that authenticating encryption verification is either Mandatory or Optional.
+* `manta.encryption_key_path` (**MANTA_ENCRYPTION_KEY_PATH**)
+The path on the local filesystem or a URI understandable by the JVM indicating the
+location of the private key used to perform client-side encryption. If this value is
+non-null, then no other encryption key values can be non-null.
+* `manta.encryption_key_bytes`
+The private key used to perform client-side encryption as a byte array. If this value is
+non-null, then no other encryption key values can be non-null.
+* `manta.encryption_key_bytes_base64` (**MANTA_ENCRYPTION_KEY_BYTES**)
+The private key used to perform client-side encryption encoded as a base64 string.
+If this value is non-null, then no other encryption key values can be non-null.
+
+Below is an example of using all of the defaults and only setting the `manta.user` and `manta.key_id`.
+
+```java
+ConfigContext defaultConfig = new DefaultsConfigContext();
+ConfigContext customConfig = new StandardConfigContext()
+    .setMantaKeyId("d4:18:cc:34:43:a8:5a:aa:76:1c:35:36:ba:08:1e:aa")
+    .setMantaUser("test-user");
+ConfigContext config = new ChainedConfigContext(defaultConfig, customConfig);
+```
+
+## Client-side Encryption
+
+In order to enable client side encryption for downloading and decrypting encrypted files, please set the following
+system properties. Please consult the [Configuration Parameters list](/INSTALL.md#parameters) for the corresponding
+environment variable.
+
+- `manta.client_encryption` - set to `true`
+- `manta.encryption_key_bytes_base64` or `manta.encryption_key_bytes`
+
+Additionally, you should set the following system properties to support encrypting and uploading files using client-side
+encryption.
+
+- `manta.encryption_algorithm`
+- `manta.encryption_key_id`
+
+Below is a table of each of the supported encryption algorithms and the features they provide.
```

Comment:
That's a crazy big question. How would you address this question:
> Where can I learn more about particle physics?

I mean this in all seriousness. Should we link out to an simple explanation page on the internet? This might give folks a false sense of confidence.

Reply:
I wouldn't say that it requires a simple explanation link-out. It is a really, really big question. Personally, I want to learn more and I think I would better be able to phrase that question if I could better articulate not just what they are but how they work.

I would assume that most people who have gotten to this point know what they need to know about encryption, but I want to know what you all find to be the most trustworthy resources on the topic. Yes, I can read the wikipedia page, but beyond that, I want to know why one may be a better choice than another. What resources can you point me to?